### PR TITLE
CVE-2014-7809

### PR DESCRIPTION
**Pull request by bnk5096**
Update CVE-2014-7809

### CODE REVIEW

File: cves/CVE-2014-7809.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ announced_instructions: |
   Was there a date that this vulnerability was announced to the world? You can
   find this in changelogs, blogs, bug reports, or perhaps the CVE data.
   Please enter your date in YYYY-MM-DD format.
-announced:
+announced: 2014-12-10
```
Comment (jehling) [inline cves/CVE-2014-7809.yml]: Apache seems to have an announcement that predates this date in accordance with their wiki at https://cwiki.apache.org/confluence/display/WW/S2-023 

- *Date:* 2014-11-21 by Lukasz Lenart

File: cves/CVE-2014-7809.yml [hunk 2]
```diff
@@ -142,7 +155,11 @@ interesting_commits:
     emerging themes?
     If there are no interesting commits, demonstrate that you completed this section
     by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: |
+    There were no commits relevant to this issue between the VCC and FIX. It was almost
+    a decade from when the issue was introduced and when it was fixed. In this time there
+    were countless commits expending features and correcting bugs, but no adjustments
+    that I could find regarding the generation of CSRF tokens.
```
Comment (jehling) [inline cves/CVE-2014-7809.yml]: Although true that there are no commits directly relevant to this vulnerability, the nature of this section is to provide context behind different things that could be happening during your vulnerability timeline. 

For example, a quick search through the history of that file in particular, looking back from the date of the Fix commit, shows a tremendous amount of activity throughout 2006 in regards to logging changes and functional refactors. Additional changes to error logging appear to resume around 2012 / 2013, which may coincide with other development goals for a given feature version. 

I recommend looking through this segment of time again and seeing if anything catches your eye. For convenience, a command I used earlier to grab data for the file in question is posted below. 

`git log --before="2014-11-21" core/src/main/java/org/apache/struts2/util/TokenHelper.java`

File: cves/CVE-2014-7809.yml [hunk 3]
```diff
@@ -174,8 +191,10 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: | 
+        This vulnerability lies in their protection from Cross Site Request Forgery
+        and this is done out of distrust of inputs.
```
Comment (jehling) [inline cves/CVE-2014-7809.yml]: I would argue the nature of this vulnerability is not `distrust inputs` but rather `frameworks are optional`. In this case, the "framework" is simply the Java language itself. That being said, the default random number generator class was used in place of a more secure option. This implies either ignorance or misunderstanding on behalf of the developers with regards to how secure Java's number generator truly is.

**Review by jehling — COMMENTED** [review]
Feedback posted. +1 Up Vote.

File: cves/CVE-2014-7809.yml (hunk 1 above)
Comment (bnk5096) [reply cves/CVE-2014-7809.yml]: Changes addressed.

File: cves/CVE-2014-7809.yml (hunk 2 above)
Comment (bnk5096) [reply cves/CVE-2014-7809.yml]: Thanks for such thorough feedback here. I appreciate it and have taken this all into consideration in my additions!

File: cves/CVE-2014-7809.yml (hunk 3 above)
Comment (bnk5096) [reply cves/CVE-2014-7809.yml]: I think I initially thought of it from the perspective of not trusting the authentication tokens, but this suggestion makes a lot more sense.

**Comment by bnk5096** [comment]
> Feedback posted. +1 Up Vote.

Upvotes added.

**Pull request merged**